### DISCUSSION TITLE
AX100 decoding

### DISCUSSION THREAD

**Discussion by xjx** [daniestevez/gr-satellites#542] [Q&A] [answered]
Hello all!

I am trying to decode IQ recordings made with AX100, using the gr-satellites AX100 deframer. 

However, there are some settings at the AX100 transmitter side that should be considered to be compatible with that deframer config.

For example, assuming that AX100 deframer block is configured as ASM+Golay, no scrambler, sync threshold 7 and no command line options, what would be the settings required at the AX100 transmitter side? 

Example settings at AX100 transmitter for a given IQ recording:
Mode 5 = ASM+Golay
csp_hmac = false
csp_rs = false
csp_crc = false
csp_rand = false
preamb = 0xAA
Are these settings in accordance with the deframer options stated above? What should be changed? is there anything else tobe considered? Is the default sync word being transmitted in accordance with the deframer syncword?

As additional information, the IQ reqs are 48ksps and the baud rate is 1200bps.

So far, I could not decode them. I can ask for new recordings made with a different tx configuration.

Thanks in advance.

**Comment by daniestevez** [comment]
Hi. Sadly the AX100 manual is not publicly available, which makes helping with this more tricky.

Some comments about your receiver configuration is that a syncword threshold of 7 is too high for a 32 bit syncword. You'll get lots of false syncword detections. I usually use 4 with 32 bit syncwords. Also, using no scrambler is generally not recommended.

Regarding the transmitter, all the settings that begin by `csp_` seem to refer to the datalink layer (CSP frames) and not to the physical layer, so they are not relevant at this point. I think that the preamble is irrelevant (as long as this setting controls what I imagine, which is a bit pattern transmitted multiple times before the actual syncword is sent). The receiver only uses this to train the clock recovery loop, so the pattern doesn't matter that much as long as there are enough bit transitions. You need to find and validate the settings that control the scrambling and the symbol rate, though.

**Reply by xjx** [reply]
HI Daniel, thank you for the quick response.

I just add this information: with syncword threshold of 7 I am still having no packets decoding (but as you mention we should be expecting false detections). So probably there is another issue happening here. I have no clues so far.

I ack the fact that csp_ settings refer to higher layers and not the physical layer. THanks.

I will use scrambler then. Regarding your comment on the bitrate, Is the 1200bps transmitted signal ok for compatibility with the deframer?

I will try to provide additional information.

**Reply by daniestevez** [reply]
> Regarding your comment on the bitrate, Is the 1200bps transmitted signal ok for compatibility with the deframer?

The deframer doesn't care about the bitrate. This is the job of the demodulator. You could use any bitrate (symbol rate) you want with the gr-satellites FSK Demodulator.

**Comment by xjx** [comment]
Daniel, for further reference i attach plots of the same signal, with different zoom levels. The TX settings are those mentioned above.
![Screenshot_20240222_095200_Chrome](https://github.com/daniestevez/gr-satellites/assets/751276/1fc3190d-c3ac-45a6-ab77-b9a37d2fafc7)
![Screenshot_20240222_095219_Chrome](https://github.com/daniestevez/gr-satellites/assets/751276/420c9483-5b16-41ca-a8f1-7d3f60eff64a)


This is the quadrature demodulated signal that is used to attack the AX100 Deframer. I would appreciate your comments. For example, does this signal look ok? What settings should I try within the AX100 deframer?

**Reply by daniestevez** [reply, marked as answer]
Are you passing this signal directly to the AX100 deframer? The deframer needs soft symbols (one sample per symbol), but this looks like something having several samples per symbol.

**Reply by xjx** [reply]
Thanks Daniel, that actually solved the problem.
